rules/TypeDeclaration/TypeAnalyzer/AlwaysStrictScalarExprAnalyzer.php

```diff
@@ -53,12 +54,43 @@ public function matchStrictScalarExpr(Expr $expr): ?Type
         }
 
         if ($expr instanceof FuncCall) {
-            return $this->resolveFuncCallType($expr);
+            $returnType = $this->resolveFuncCallType($expr);
+
+            if ($returnType === null) {
+                return null;
+            }
+
+            if (! $this->isScalarType($returnType)) {
+                return null;
+            }
+
+            return $returnType;
         }
 
         return null;
     }
 
+    private function isScalarType(Type $type): bool
+    {
+        if ($type instanceof StringType && ! $type instanceof ConstantStringType) {
+            return true;
+        }
+
+        if ($type instanceof FloatType) {
+            return true;
+        }
+
+        if ($type instanceof IntegerType) {
+            return true;
+        }
+
+        if ($type instanceof BooleanType) {
+            return true;
+        }
+
+        return false;
```

Comment:
```suggestion
        if (! $type instanceof ScalarType) {
            return false;
        }

        return $this->resolveTypeFromScalar($type) instanceof Type;
```

Reply:
This always returns false because the `$type` is `PHPStan\Type\Type` and that is never `PhpParser\Node\Scalar`.

Reply:
Ok 👍